pkg/scheduler/util/scheduler_helper.go

```diff
@@ -237,3 +247,17 @@ func ValidateVictims(preemptor *api.TaskInfo, node *api.NodeInfo, victims []*api
 	}
 	return nil
 }
+
+// ResourceReservation is struct used for resource reservation
+type ResourceReservation struct {
+	TargetJob   *api.JobInfo
+	LockedNodes map[string]*api.NodeInfo
+}
+
+// NewResourceReservation is used to create global instance
+func NewResourceReservation() *ResourceReservation {
+	return &ResourceReservation{
+		TargetJob:   nil,
+		LockedNodes: map[string]*api.NodeInfo{},
+	}
+}
```

Comment:
There is another question here, why not put the `ResourceReservation` information in the session

Reply:
> There is another question here, why not put the `ResourceReservation` information in the session?

Good question! In the design, ResourceReservation must be shared accross session.